src/Lib.P2P/SecureCommunication/Secio1.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Logging;
+using Lib.P2P.Cryptography;
+using Lib.P2P.Protocols;
+using MultiFormats;
+using Org.BouncyCastle.Security;
+using ProtoBuf;
+using Semver;
+
+namespace Lib.P2P.SecureCommunication
+{
+    /// <summary>
+    ///   Creates a secure connection with a peer.
+    /// </summary>
+    public class Secio1 : IEncryptionProtocol
+    {
+        private static ILog log = LogManager.GetLogger(typeof(Secio1));
+
+        /// <inheritdoc />
+        public string Name { get; } = "secio";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 0);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            await EncryptAsync(connection, cancel).ConfigureAwait(false);
+        }
+
+#pragma warning disable VSTHRD103 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)